research/python/src/features/hedge_pressure_research.py

```
@@ -0,0 +1,406 @@
+    Returns:
+        HedgePressureResult with pressure metric and metadata
+
+    Raises:
+        ValueError: If input data is invalid
+    """
+    if config is None:
+        config = HedgePressureConfig()
+
+    # Input validation
+    if not isinstance(calls_oi, dict) or not isinstance(puts_oi, dict):
+        raise ValueError("OI data must be dictionaries mapping strike to quantity")
```

Ruff (0.14.10)

Prefer `TypeError` exception for invalid type

(TRY004)